Sources/XCTest/XCTestCase.swift

```diff
@@ -26,7 +26,11 @@ public typealias XCTestCaseEntry = (testCaseClass: XCTestCase.Type, allTests: [(
 
 public class XCTestCase {
 
-    public required init() {
+    /// The name of the test case, consisting of its class name and the method name it will run
+    public let name: String
+
+    public required init(methodName: String) {
```

Comment:
Apple XCTest's `XCTestCase` defines two initializers: `-[XCTestCase initWithInvocation:]` and `-[XCTestCase testCaseWithSelector]`. I suppose neither of those could be made available in an environment without the Objective-C runtime. But I feel like we shouldn't provide `public` API beyond what Apple XCTest defines--that's the stuff evolution proposals are for. (I feel the empty initializer is fine because Apple XCTest also technically allows you to call `-[[XCTestCase alloc] init]`).

Is there a reason this API needs to be `public`? It seems we could have `methodName` be a `private` attribute that can be set after the `XCTestCase` instance is initialized. That would avoid the issue of providing an additional `public` API.